pkg/controllers/updaterun/controller.go

```diff
@@ -158,9 +158,12 @@ func (r *Reconciler) Reconcile(ctx context.Context, req runtime.Request) (runtim
 		return runtime.Result{}, r.recordUpdateRunSucceeded(ctx, updateRun)
 	}
 
-	// Execute the updateRun.
-	if state == placementv1beta1.StateRun {
-		klog.V(2).InfoS("Continue to execute the updateRun", "state", state, "updatingStageIndex", updatingStageIndex, "updateRun", runObjRef)
+	switch state {
```

Comment:
Not able to comment inline, so put the comments here:
1. On line 116 where you check if the updateRun has been initialized, now you can use the new util function instead.
2. On line 128 where you check if the updateRun has finished, you still check against generation. Looks not correct?
3. Do we ever update the generations to keep up-to-date?